File: client/src/repository/todo_repository.ts
import { Form } from '@/types/todo'
import Todo from '@/models/todo'

export default interface TodoRepository {
    Get(): Promise<Todo[]>;
    Create(form: Form): Promise<string>;
    Update(id: number, form: Form): Promise<string>;
    Delete(id: number): Promise<string>;
}

File: api/repository/todo-repository.go
package repository

import (
	"context"
	"errors"
	"go-vue-todo/model"
)

var ErrEntityNotFound = errors.New("entity not found")

type TodoRepository interface {
	List(context.Context, string) ([]*model.Todo, error)
	GetById(context.Context, int) (*model.Todo, error)
	Create(context.Context, *model.Todo) error
	Update(context.Context, *model.Todo) error
	Delete(context.Context, int) error
}